### PR TITLE
OMG-221 - refactor: make omg_jsonrpc not dependent on omg_api

### PR DESCRIPTION
**Pull request by pdobacz**
co-authored with @pgebal during refactor Friday

closes OMG-221

### CODE REVIEW

File: apps/omg_jsonrpc/lib/server.ex [hunk 1]
```diff
@@ -14,13 +14,16 @@
...
   @moduledoc """
-  Exposes OMG.API via jsonrpc 2.0 over HTTP. It leverages the generic OMG.JSONRPC.Exposer convenience module
+  Exposes an API via jsonrpc 2.0 over HTTP. It leverages the generic OMG.JSONRPC.Exposer convenience module
 
   Only handles the integration with the JSONRPC2 package
   """
   use JSONRPC2.Server.Handler
 
+  # Running :omg_jsonrpc requires setting this environmental variable
+  @api_module Application.fetch_env!(:omg_jsonrpc, :api_module)
```
Comment (paulperegud) [inline apps/omg_jsonrpc/lib/server.ex]: I believe this one is broken. I.e. Application.fetch_env is supposed to work in runtime, not in compile time. This effectively reads value from files that exist on dev machine during compilation (mix.exs or config.exs) and will ignore whatever way we will use to provide configuration to packaged release.

Comment (paulperegud) [reply apps/omg_jsonrpc/lib/server.ex]: You can test it by running `iex -S mix run --no-start` and changing the value of env variable via :application.set_env

File: apps/omg_performance/lib/performance.ex [hunk 2]
```diff
@@ -158,6 +158,7 @@ defmodule OMG.Performance do
 
     Application.put_env(:omg_eth, :contract_addr, contract_addr)
 
+    # FIXME
```
Comment (paulperegud) [inline apps/omg_performance/lib/performance.ex]: Somehow this line does not trigger credo, while it should.

File: apps/omg_jsonrpc/lib/server.ex (hunk 1 above)
Comment (pdobacz) [reply apps/omg_jsonrpc/lib/server.ex]: actually, we gave it a moment of thought and realised, we want it to be a compile-time constant in fact. Does this make it easier to live with?

File: apps/omg_performance/lib/performance.ex (hunk 2 above)
Comment (pdobacz) [reply apps/omg_performance/lib/performance.ex]: apparently `# FIXME a` works, I'll file an issue to `credo`

File: apps/omg_jsonrpc/lib/server.ex (hunk 1 above)
Comment (paulperegud) [reply apps/omg_jsonrpc/lib/server.ex]: Yeah, this works fine. Commenting that such usage is intended would be good to make reading simpler.

**Pull request merged**